### PR TITLE
Make native builds work out-of-the-box with H2

### PR DESCRIPTION
**Pull request by mraible**
Fixes https://github.com/jhipster/generator-jhipster-quarkus/issues/172.

### CODE REVIEW

**Comment by mraible** [comment]
@avdev4j The `web.xml` causes issues when doing a native build because it references H2 classes. I'm not sure this file is needed since there's to DB console link in the UI. If I try to go to `/h2-console`, I get a 404. 

If we need to add it back in, we should figure out a way to include it in `dev` and remove it in `prod`.

**Comment by mraible** [comment]
I've confirmed that `./mvnw package -Pnative` works out-of-the-box for the following `.yo-rc.json`:

```json
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.mycompany.myapp",
      "nativeLanguage": "en"
    },
    "jhipsterVersion": "6.10.5",
    "applicationType": "monolith",
    "baseName": "jhipster",
    "packageName": "com.mycompany.myapp",
    "packageFolder": "com/mycompany/myapp",
    "serverPort": "8080",
    "authenticationType": "oauth2",
    "cacheProvider": "no",
    "enableHibernateCache": false,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "mysql",
    "messageBroker": false,
    "buildTool": "maven",
    "embeddableLaunchScript": false,
    "useSass": true,
    "clientPackageManager": "npm",
    "clientFramework": "angularX",
    "clientTheme": "none",
    "clientThemeVariant": "",
    "creationTimestamp": 1614785275213,
    "jhiPrefix": "jhi",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "otherModules": [
      {
        "name": "generator-jhipster-quarkus",
        "version": "1.1.0"
      }
    ],
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": ["en"],
    "blueprints": [
      {
        "name": "generator-jhipster-quarkus",
        "version": "1.1.0"
      }
    ],
    "searchEngine": false,
    "serviceDiscoveryType": false
  }
}
```

However, if you try to run it, you'll get the following errors:

```
$ target/*runner
Both 'auth-server-url' and 'client-id' properties must be configured
```

If I use the Okta CLI and run `okta .okta.env`, it works. This makes me think that Keycloak is not configured for prod by default. FWIW, the main generator has Keycloak set for `dev` and `prod`.

File: generators/server/templates/quarkus/pom.xml.ejs [hunk 1]
```diff
@@ -170,6 +170,12 @@
             <version>${camel-quarkus-jackson.version}</version>
         </dependency>
         <%_ } _%>
+        <%_ if (!skipUserManagement) { _%>
+        <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-lang3</artifactId>
```
Comment (danielpetisme) [inline generators/server/templates/quarkus/pom.xml.ejs:176]: Why do we need commons-lang?

File: generators/server/templates/quarkus/src/main/resources/application.properties.ejs [hunk 2]
```diff
@@ -68,7 +68,9 @@ quarkus.swagger-ui.always-include=true
 quarkus.swagger-ui.path=/swagger-ui
 
 quarkus.native.additional-build-args=\
+<%_ if (authenticationType !== 'oauth2') { _%>
```
Comment (danielpetisme) [inline generators/server/templates/quarkus/src/main/resources/application.properties.ejs:71]: Given the fact `resources-config.json` is specific to JWT (basically it explicit the need to include the privateKey for token signature) could you use the condition`authenticationType === 'jwt'`
please

File: test/server.spec.js [hunk 3]
```diff
@@ -48,21 +47,6 @@ describe('Subgenerator server of quarkus JHipster blueprint', () => {
         it('contains hibernate second level cache needle', () => {
             assert.fileContent(`${SERVER_MAIN_RES_DIR}application.properties`, '# jhipster-quarkus-needle-hibernate-cache-add-entry');
         });
-
-        it('contains h2 servlet data in web.xml', () => {
```
Comment (danielpetisme) [inline test/server.spec.js:52]: @avdev4j  Do you remember why we needed the web.xml file?
It seems that the h2 console is not available.

**Review by danielpetisme — CHANGES_REQUESTED** [review]

File: generators/server/templates/quarkus/pom.xml.ejs (hunk 1 above)
Comment (mraible) [reply generators/server/templates/quarkus/pom.xml.ejs:176]: If I don't add it, a CI build fails when trying to compile `RandomUtil.java` that imports it. My guess is that the Quarkus H2 dependency was pulling it in before.

Comment (danielpetisme) [reply generators/server/templates/quarkus/pom.xml.ejs:176]: Ok. let it as is.
I'll tend to remove it later on.
I'm not a fan of pulling a full dependency for basically one class...

**Comment by avdev4j** [comment]
I'm a bit worried because it was working as expected :D 
Screenshot 2021-03-04 at 15 11 21

Maybe we should find a way to make it works with native build instead of removing it at all

**Comment by mraible** [comment]
I only tried `/h2-console/`, not with the `/login.jsp` suffix. Here's the `.yo-rc.json` you can test with.

```
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.mycompany.myapp",
      "nativeLanguage": "en"
    },
    "jhipsterVersion": "6.10.5",
    "applicationType": "monolith",
    "baseName": "jhipster",
    "packageName": "com.mycompany.myapp",
    "packageFolder": "com/mycompany/myapp",
    "serverPort": "8080",
    "authenticationType": "oauth2",
    "cacheProvider": "no",
    "enableHibernateCache": true,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "mysql",
    "messageBroker": false,
    "buildTool": "maven",
    "embeddableLaunchScript": false,
    "useSass": true,
    "clientPackageManager": "npm",
    "clientFramework": "angularX",
    "clientTheme": "none",
    "clientThemeVariant": "",
    "creationTimestamp": 1614834465776,
    "jhiPrefix": "jhi",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "otherModules": [
      {
        "name": "generator-jhipster-quarkus",
        "version": "1.1.1"
      }
    ],
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": ["en"],
    "blueprints": [
      {
        "name": "generator-jhipster-quarkus",
        "version": "1.1.1"
      }
    ]
  }
}
```

With 1.1.1, you can generate a new app with this, and run `./mvnw package -Pnative`. With 1.1.0, the native build doesn't work.

**Comment by avdev4j** [comment]
I have just clicked on the db menu in UI.

Maybe the problem comes from the compatibility with oauth2 and this web.xml config.

In this case, let's drop oauth2! (joke)

I will take a look, with this .yo-rc.json and I will try to add a new workflow for testing native compilation and avoid this kind of issue in the future.

**Comment by mraible** [comment]
For me, there is no DB menu in the UI to click on.

The problem from the native build is related to H2. Since the web.xml references H2 classes, it's part of the problem.

Another thing I noticed with the native build is it sends `0.0.0.0` instead of `localhost`.

```
2021-03-04 09:57:19,067 INFO  [io.quarkus] (main) jhipster 1.0.0-SNAPSHOT native 
(powered by Quarkus 1.12.1.Final) started in 2.001s. Listening on: http://0.0.0.0:8080
```

This breaks OAuth because this hasn't been registered as redirect URI.

**Pull request merged**